ext/compressors/lz4/lz4_compress.c

```diff
@@ -214,6 +214,18 @@ lz4_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, siz
         decoded = LZ4_decompress_safe((const char *)src + sizeof(LZ4_PREFIX), (char *)dst,
           (int)prefix.compressed_len, (int)dst_len);
 
+    /*
+     * In the majority of cases, when the decompressor successfully decompresses the entire source
+     * data, the decoded bytes matches the uncompressed length. However, there are some situations
+     * where the data is corrupted and the decompressor is unable to decompress the entire source
+     * data, in which case it differs and returns an error.
+     */
+    if (decoded != (int)prefix.uncompressed_len) {
```

Comment:
Should we check against useful_len or uncompressed_len?

I think we should put this check inside if (decoded >= 0) section and cast decoded to (uint32_t)

Reply:
@quchenhao I verified with the test run that `useful_len` and `uncompressed_len` are always the same, but as per the comments [here](https://github.com/wiredtiger/wiredtiger/blob/develop/ext/compressors/lz4/lz4_compress.c#L74) I am more inclined towards comparing against `prefix.useful_len`.